clients/http_json_rpc_client.go

```diff
@@ -181,6 +191,28 @@ func (c *GenericHttpJsonRpcClient) shutdown() {
 	c.processBatch(true)
 }
 
+func (c *GenericHttpJsonRpcClient) getHttpClient() *http.Client {
+	proxyPoolRegistry, err := NewProxyPoolRegistry(common.GetConfig().ProxyPools, c.logger)
```

Comment:
This seems to be running on 'every request' so it's not optimal to create a new pool registry everytime.

The better approach is to create the registry once during Client creation/init, and then here just to "access" it.